examples/octet_stream/text-client.cpp

```diff
@@ -0,0 +1,144 @@
+  auto conn
+    = caf::net::octet_stream::with(sys)
+        // Optionally enable TLS.
+        .context(ssl::context::enable(use_ssl)
+                   .and_then(ssl::emplace_server(ssl::tls::v1_2))
+                   .and_then(ssl::load_verify_file_if(ca_file)))
+        // Connect to "$host:$port".
+        .connect(host, port)
+        // If we don't succeed at first, try up to 10 times with 1s delay.
+        .retry_delay(1s)
+        .max_retry_count(9)
+        // After connecting, spin up a worker that prints received inputs.
+        .start([&sys](auto pull, auto push) {
+          sys.spawn([pull](caf::event_based_actor* self) {
+            pull.observe_on(self)
+              .transform(caf::flow::byte::split_as_utf8_at('\n'))
+              .do_on_error([self](const caf::error& err) {
+                self->println("*** connection error: {}", err);
+              })
+              .do_finally([self] {
+                self->println("*** lost connection to server -> quit\n*** use "
+                              "CTRL+D or CTRL+C to terminate");
+                self->quit();
+              })
+              .for_each([](const caf::cow_string& line) {
+                std::cout << "reply: " << line.str() << std::endl;
+              });
+          });
+          // Spin up a second worker that reads from std::cin and sends each
+          // line to the server. Put that to its own thread since it's doing
+          // blocking I/O calls.
+          sys.spawn<caf::detached>([&sys, push] {
+            auto lines = caf::async::make_blocking_producer(push);
+            assert(lines);
+            auto line = std::string{};
+            while (std::getline(std::cin, line)) {
+              std::vector<std::string> parsed_line;
+              caf::split(parsed_line, line, " ");
+              if (parsed_line.size() < 2) {
+                sys.println("Invalid command");
+                continue;
+              }
+              std::string json_line = R"_({"type":")_" + parsed_line[0]
+                                      + R"_(", "key":")_" + parsed_line[1]
+                                      + R"_(")_";
+              if (parsed_line.size() == 3) {
+                json_line += R"_(, "value":")_" + parsed_line[2] + R"_(")_";
+              }
+              json_line += "}\n";
+              lines->push(caf::as_bytes(caf::make_span(json_line)));
+              line.clear();
+            }
+          });
+        });
+  ;
+  // Report any error to the user.
+  if (!conn) {
+    sys.println("*** unable to run at port {}: {}", port, conn.error());
+    return EXIT_FAILURE;
+  }
+  // Wait for CTRL+C or SIGTERM.
+  sys.println("*** Client is running, press CTRL+C to stop");
+  while (!shutdown_flag)
+    std::this_thread::sleep_for(250ms);
+  // Restore the default handlers.
+  signal(SIGTERM, default_term);
+  signal(SIGINT, default_sint);
+  // Shut down the client.
+  sys.println("*** shutting down");
+  return EXIT_SUCCESS;
```

Comment:
Actually, the actor reading from `cin` can never terminate this way since it won't ever check it's mailbox. Pull out out the `cin` loop from the actor and do the loop in `main` instead, pushing data into the actor.